### PR TITLE
IGNITE-19831 Rename RaftManager methods to better reflect their behavior

### PR DESCRIPTION
**Pull request by sashapolo**
https://issues.apache.org/jira/browse/IGNITE-19831

### CODE REVIEW

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/OnRevisionAppliedCallback.java [hunk 1]
```diff
@@ -29,7 +29,7 @@ public interface OnRevisionAppliedCallback {
     /**
      * Notifies of completion of processing of Meta Storage watches for a particular revision.
      *
-     * @param watchEvent Event with modified Meta Storage entries processed at least one Watch.
```
Comment (tkalkirill) [inline modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/OnRevisionAppliedCallback.java:32]: According to the ticket description, this does not need to be corrected in this ticket.

File: modules/raft-api/src/main/java/org/apache/ignite/internal/raft/RaftManager.java [hunk 2]
```diff
@@ -26,61 +26,91 @@
 
 /**
  * Raft manager.
+ *
+ * <p>This class contains two groups of methods for starting Raft nodes: {@code #startRaftGroupNode} and
+ * {@code startRaftGroupNodeAndWaitNodeReadyFuture} (and its overloads). When using {@code #startRaftGroupNode}, Raft log re-application
```
Comment (tkalkirill) [inline modules/raft-api/src/main/java/org/apache/ignite/internal/raft/RaftManager.java:31]: Give an example when it is not necessary to wait for reapplying the log.

File: modules/raft-api/src/main/java/org/apache/ignite/internal/raft/RaftManager.java [hunk 3]
```diff
@@ -26,61 +26,91 @@
...
+ * does not get performed and external synchronisation methods must be used to avoid observing a Raft node in inconsistent state. The other
+ * group of methods synchronously waits for the Raft log to be re-applied, so no external synchronisation is required.
  */
 public interface RaftManager extends IgniteComponent {
+    /**
+     * Starts a Raft group and a Raft service on the current node, using the given service factory.
```
Comment (tkalkirill) [inline modules/raft-api/src/main/java/org/apache/ignite/internal/raft/RaftManager.java:37]: Add to the description that it doesn't wait for the raft log to be applied.

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/OnRevisionAppliedCallback.java (hunk 1 above)
Comment (sashapolo) [reply modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/OnRevisionAppliedCallback.java:32]: And what do you propose? To remove this change?

Comment (tkalkirill) [reply modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/OnRevisionAppliedCallback.java:32]: yep

Comment (sashapolo) [reply modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/OnRevisionAppliedCallback.java:32]: Why?

**Pull request merged**